x-pack/plugins/cases/server/services/attachments/index.ts

```diff
@@ -251,12 +250,11 @@ export class AttachmentService {
       );
 
       const soExtractor = getAttachmentSOExtractor(res.attributes);
-      const so = soExtractor.populateFieldsFromReferences<AttachmentAttributes>(res);
-      const injectedAttributes = injectPersistableReferencesToSO(so.attributes, so.references, {
-        persistableStateAttachmentTypeRegistry: this.persistableStateAttachmentTypeRegistry,
-      });
-
-      return { ...so, attributes: { ...so.attributes, ...injectedAttributes } };
+      return injectAttachmentSOReferences(
```

Comment:
nit: sorry I know I suggested this haha, but could we rename it to `injectAttachmentSOAttributesFromRefs` or something similar to communicate that we are creating attributes from references.

The current name makes me think we are injecting references, instead we're injecting attributes.